pkg/helm/helm_test.go

```diff
@@ -213,6 +308,80 @@ func TestPullCommand(t *testing.T) {
 				"suse-edge/kubevirt",
 			},
 		},
+		{
+			name: "HTTP repository with auth and skip TLS verify",
+			repo: &image.HelmRepository{
+				Name: "suse-edge",
+				URL:  "https://suse-edge.github.io/charts",
+				Authentication: image.HelmAuthentication{
+					Username: "user",
+					Password: "pass",
+				},
+				SkipTLSVerify: true,
+			},
+			chart: "kubevirt",
+			expectedArgs: []string{
+				"helm",
+				"pull",
+				"suse-edge/kubevirt",
+				"--insecure-skip-tls-verify",
+			},
+		},
+		{
+			name: "HTTP repository with auth and plain HTTP",
+			repo: &image.HelmRepository{
+				Name: "suse-edge",
+				URL:  "http://suse-edge.github.io/charts",
+				Authentication: image.HelmAuthentication{
+					Username: "user",
+					Password: "pass",
+				},
+				PlainHTTP: true,
+			},
+			chart: "kubevirt",
+			expectedArgs: []string{
+				"helm",
+				"pull",
+				"suse-edge/kubevirt",
+				"--plain-http",
+			},
+		},
+		{
+			name: "OCI repository with auth and skip TLS verify",
+			repo: &image.HelmRepository{
+				Name: "apache-repo",
+				URL:  "oci://registry-1.docker.io/bitnamicharts/apache",
+				Authentication: image.HelmAuthentication{
+					Username: "user",
+					Password: "pass",
+				},
+				SkipTLSVerify: true,
+			},
+			expectedArgs: []string{
+				"helm",
+				"pull",
+				"oci://registry-1.docker.io/bitnamicharts/apache",
+				"--insecure-skip-tls-verify",
+			},
+		},
+		{
+			name: "OCI repository with auth and plain HTTP",
+			repo: &image.HelmRepository{
+				Name: "apache-repo",
+				URL:  "oci://registry-1.docker.io/bitnamicharts/apache",
+				Authentication: image.HelmAuthentication{
+					Username: "user",
+					Password: "pass",
+				},
+				PlainHTTP: true,
+			},
+			expectedArgs: []string{
+				"helm",
+				"pull",
+				"oci://registry-1.docker.io/bitnamicharts/apache",
+				"--plain-http",
```

Comment:
Pardon my ignorance - does this even make sense? Isn't the OCI protocol completely separate?

Reply:
I can’t say I fully understand it just yet but from what I understand at the moment is that it still uses HTTP(s) under the hood:
```
curl -u "user:root" "http://192.168.1.185:5000/v2/_catalog"
{"repositories":["apache/apache","rancher/rancher"]}
```

```
08:07:10 danny@ubuntu ~ → helm registry login http://192.168.1.185:5000
Username: user
Password:
INFO[0002] Error logging in to endpoint, trying next endpoint  error="Get \"https://192.168.1.185:5000/v2/\": http: server gave HTTP response to HTTPS client"
Error: Get "https://192.168.1.185:5000/v2/": http: server gave HTTP response to HTTPS client

08:07:27 danny@ubuntu ~ → helm registry login http://192.168.1.185:5000 --insecure
Username: user
Password:
WARN[0001] insecure registry http://192.168.1.185:5000 should not contain 'http://' and 'http://' has been removed from the insecure registry config
INFO[0001] Error logging in to endpoint, trying next endpoint  error="Get \"https://192.168.1.185:5000/v2/\": http: server gave HTTP response to HTTPS client"
Login Succeeded
```